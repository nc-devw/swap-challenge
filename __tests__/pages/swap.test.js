import { render } from "@testing-library/react";
import Swap, { getServerSideProps } from "@/pages/swap";

jest.mock("../../helpers/fetchData", () => () => ({ data: [] }));

describe("Swap", () => {
  it("should render correctly", () => {
    const { asFragment } = render(<Swap coins={[]} />);
    expect(asFragment()).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it("should receive properties correctly", async () => {
    const props = await getServerSideProps();
    expect(props).toEqual({
      props: { coins: [] },
    });
  });
});
