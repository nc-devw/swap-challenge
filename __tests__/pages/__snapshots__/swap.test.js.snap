// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swap should render correctly 1`] = `[Function]`;
